homeassistant/components/moehlenhoff_alpha2/climate.py

```diff
@@ -0,0 +1,152 @@
+        )
+
+    def _handle_heatarea_data_updated(self, data):
+        """Handle updated heatarea data."""
+        if data["NR"] == self._data["NR"]:
+            self._data = data
+            self.async_schedule_update_ha_state()
+
+    @property
+    def name(self):
+        """Return the name of the climate device."""
+        return self._data["HEATAREA_NAME"]
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return float(self._data.get("T_TARGET_MIN", 0))
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        return float(self._data.get("T_TARGET_MAX", 30))
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return float(self._data.get("T_ACTUAL", 0))
+
+    def set_hvac_mode(self, hvac_mode: str):
+        """Set new target hvac mode."""
+        pass
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return float(self._data.get("T_TARGET", 0))
+
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperatures."""
+        target_temperature = kwargs.get(ATTR_TEMPERATURE)
+        if target_temperature is None:
+            return False
+
+        try:
+            await self._base_update_handler.async_set_target_temperature(
+                self._data["ID"], target_temperature
+            )
+            self._data["T_TARGET"] = target_temperature
+            return True
+        except Exception as update_err:  # pylint: disable=broad-except
+            _LOGGER.error("Setting target temperature failed: %s", update_err)
+            return False
+
+    @property
+    def preset_mode(self):
+        """Return the current preset mode."""
+        # HEATAREA_MODE: 0=Auto, 1=Tag, 2=Nacht
+        if self._data["HEATAREA_MODE"] == 1:
+            return PRESET_COMFORT
+        if self._data["HEATAREA_MODE"] == 2:
+            return PRESET_ECO
+        return PRESET_NONE
+
+    async def async_set_preset_mode(self, preset_mode):
+        """Set new operation mode."""
+        # HEATAREA_MODE: 0=Auto, 1=Tag, 2=Nacht
+        heatarea_mode = 0
+        if preset_mode == PRESET_COMFORT:
+            heatarea_mode = 1
+        elif preset_mode == PRESET_ECO:
+            heatarea_mode = 2
+
+        try:
+            await self._base_update_handler.async_set_heatarea_mode(
+                self._data["ID"], heatarea_mode
+            )
+            self._data["HEATAREA_MODE"] = heatarea_mode
+            if heatarea_mode == 1:
+                self._data["T_TARGET"] = self._data["T_HEAT_DAY"]
+            elif heatarea_mode == 2:
+                self._data["T_TARGET"] = self._data["T_HEAT_NIGHT"]
+            return True
+        except Exception as update_err:  # pylint: disable=broad-except
+            _LOGGER.error("Setting target temperature failed: %s", update_err)
+            return False
```

Comment:
- `async_set_preset_mode` should not return
- there is too much code in the `try` block
- `Exception` is probably too broad
    - It may be that hass can't connect to the device, log a nicer error in this case instead of `update_err`

Reply:
Okay, comitted some changes.